### PR TITLE
Add token amount

### PR DESCRIPTION
**Pull request by GHesericsu**
converted availableLiquidity from BN to TokenAmount

### CODE REVIEW

File: src/pools/reserve.ts [hunk 1]
```diff
@@ -204,7 +204,7 @@ export class JetReserve {
       pythOracle.priceData,
       pythOracle.productData,
       mintInfo.data,
-      new BN(availableLiquidity)
+      new TokenAmount(new BN(availableLiquidity), parseMintAccount(mintInfo.data).decimals, data.tokenMint)
```
Comment (Tristyn) [inline src/pools/reserve.ts:207]: You are parsing the mint twice, once in this method and once in decodeReserveData (Same decoding the mint twice in loadMultiple()). I recommend you do each step once..
```suggestion
      new BN(availableLiquidity)
```

File: src/pools/reserve.ts [hunk 2]
```diff
@@ -271,7 +271,11 @@ export class JetReserve {
         pythOracle.priceData,
         pythOracle.productData,
         multipleMintInfo[i],
-        vaults[i].amount
+        new TokenAmount(
+          new BN(vaults[i].amount.toNumber()),
+          parseMintAccount(multipleMintInfo[i]).decimals,
+          reserveInfo.tokenMint
+        )
```
Comment (Tristyn) [inline src/pools/reserve.ts:278]:
```suggestion
        vaults[i].amount
```

File: src/pools/reserve.ts [hunk 3]
```diff
@@ -310,7 +314,7 @@ export class JetReserve {
       pythOracle.priceData,
       pythOracle.productData,
       mintInfo.data,
-      new BN(availableLiquidity)
+      new TokenAmount(new BN(availableLiquidity), parseMintAccount(mintInfo.data).decimals, this.data.tokenMint)
```
Comment (Tristyn) [inline src/pools/reserve.ts:317]:
```suggestion
      new BN(availableLiquidity)
```

File: src/pools/reserve.ts [hunk 4]
```diff
@@ -335,7 +339,7 @@ export class JetReserve {
     priceData: PriceData,
     productData: ProductData,
     mintData: Buffer,
-    availableLiquidity: BN
+    availableLiquidity: TokenAmount
```
Comment (Tristyn) [inline src/pools/reserve.ts:342]:
```suggestion
    availableLiquidity: BN
```

File: src/pools/reserve.ts [hunk 5]
```diff
@@ -335,7 +339,7 @@ export class JetReserve {
...
   ) {
     const mint = parseMintAccount(mintData)
```
Comment (Tristyn) [inline src/pools/reserve.ts:344]: .

Comment (Tristyn) [inline src/pools/reserve.ts:344]:
```suggestion
    availableLiquidityLamports: BN
  ) {
    const mint = parseMintAccount(mintData)
    const availableLiquidity = new TokenAmount(availableLiquidity, mint.decimals, data.tokenMint)
    const state = ReserveStateStruct.decode(new Uint8Array(data.state))
```

**Review by Tristyn — CHANGES_REQUESTED** [review]
This is great dude! Awesome bug fixes

Looks like you're deserializing the mint then calling a function that deserializes it again.
Remember the DRY principle. Don't Repeat Yourself

**Review by callensm — APPROVED** [review]

**Review by Tristyn — APPROVED** [review]

**Pull request merged**